### PR TITLE
Add classes option

### LINKED ISSUE #20: Feature request: allow users to specify class name on any element (without BEM)
Hi Kitty,

Thanks so much for this plugin! 

I think it would be nice if the plugin exposed an API for adding a custom class name (or multiple class names) to any one of the individual elements that it renders, like the `ol`, `li`, or anchors. That way, it's more flexible and less opinionated out of the box.

For example, in my use case, I have a custom utility class that applies some standard styling to all lists. I want to be able to apply this class to the `ol` itself, but it's not currently possible to do this because the plugin strictly enforces BEM.

Here's how `markdown-it-toc-done-right` does things, just as an example:

```js
.use(markdownItTocDoneRight, {
    listClass: 'list toc-list',
    itemClass: 'toc-item',
    linkClass: 'toc-link',
    listType: 'ol',
  })
```

The benefit of this is that you can still enforce BEM yourself while also including additional classes as needed.

### PR DESCRIPTION
**Pull request by AleksandrHovhannisyan**
Fixes #20 

I can rebase this PR if you want so that there aren't too many commits. The individual commits should summarize what changed.

I hope you're okay with me making small changes, like adding a `test:watch` script or changing some comments to be jsDoc (for intellisense hints in VS Code). If not, we can revert these.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -8,23 +10,30 @@
...
-  // @param {string} content - Footnote reference content
-  // @param {string} id - Footnote id
-  // @param {string} description - Actual footnote content
+  const { 
+    baseClass = 'Footnotes',
+    title = 'Footnotes',
+    titleId = 'footnotes-label',
+    backLinkLabel = ((_, index) => `Back to reference ${index + 1}`),
+    classes = {}
+  } = options;
```
Comment (AleksandrHovhannisyan) [inline index.js:29]: We can revert this; I just found it a little more compact since you can see the defaults at a glance

File: package.json [hunk 2]
```diff
@@ -29,5 +30,8 @@
   },
   "jest": {
     "testEnvironment": "jsdom"
+  },
+  "dependencies": {
+    "clsx": "^1.1.1"
```
Comment (AleksandrHovhannisyan) [inline package.json:35]: Super tiny dependency, but I understand if you don't want the package to have any. We can create our own helper instead, just let me know.

Comment (KittyGiraudel) [reply package.json:35]: I’ve gone through the dependency and it looks very neat and tiny, so I’m all fine with keeping it. :)

File: package.json [hunk 3]
```diff
@@ -4,7 +4,8 @@
   "description": "11ty plugin to render accessible footnotes",
   "main": "index.js",
   "scripts": {
-    "test": "jest"
+    "test": "jest",
+    "test:watch": "jest --watch"
```
Comment (KittyGiraudel) [inline package.json:8]: I’m not opposed to it. I would personally just do `npm test -- --watch`, but I’m fine with having a custom script. :)

File: test.js [hunk 4]
```diff
@@ -150,4 +150,53 @@ describe('The `eleventy-plugin-footnotes` plugin', () => {
 
     expect(root.querySelector('a').getAttribute('aria-label')).toBe('Go to 1')
   })
+
+  it('should allow customizing class names for individual elements', () => {
```
Comment (KittyGiraudel) [inline test.js:154]: Thank you for adding tests!

File: index.js [hunk 5]
```diff
@@ -93,14 +103,14 @@ module.exports = (config, options = {}) => {
...
 function attrs(object) {
   return Object.keys(object).reduce((acc, key, index) => {
     return [acc, `${key}="${object[key]}"`].filter(Boolean).join(' ')
   }, '')
 }
 
-// Small utility to append element suffix to BEM block base class
-function getClass(block) {
+/** Small utility to append element suffix to BEM block base class */
+function getBemClass(block) {
```
Comment (KittyGiraudel) [inline index.js:114]: Makes sense. 👍

File: index.js (hunk 1 above)
Comment (KittyGiraudel) [reply index.js:29]: Nah, it’s okay. I’m fine with that since it’s semantically equivalent. :)

**Review by KittyGiraudel — APPROVED** [review]
It’s looking all good! I think we can just merge and publish. :)

**Comment by AleksandrHovhannisyan** [comment]
Cool, sounds good! 😄

**Comment by KittyGiraudel** [comment]
Released in https://github.com/KittyGiraudel/eleventy-plugin-footnotes/releases/tag/0.10.0. Thank you so much!

**Pull request merged**